src/AcmTable/AcmTable.tsx

```diff
@@ -605,22 +616,54 @@ export function AcmTable<T>(props: AcmTableProps<T>) {
         [filtered, rows, keyFn]
     )
 
-    const actions = rowActions.map((rowAction) => {
-        return {
-            title: rowAction.title,
-            onClick: (_event: React.MouseEvent, rowId: number, rowData: IRowData) => {
-                if (groupFn || addSubRows) {
-                    const tableItem =
-                        rowData.props?.key && sorted.find((tableItem) => tableItem.key === rowData.props.key)
-                    if (tableItem) {
-                        rowAction.click(tableItem.item)
+    // Function to parse provided actions from AcmTable IAcmRowAction --> Patternfly Table IAction
+    const parseRowAction = (rowActions: IAcmRowAction<T>[]) => {
+        const actions: IAction[] = []
+        rowActions.forEach((action) => {
+            // Add separator if specified
+            if (action.addSeparator) {
+                actions.push({
+                    isSeparator: true,
+                })
+            }
+            // Add row action
+            actions.push({
+                title: action.title,
+                onClick: (_event: React.MouseEvent, rowId: number, rowData: IRowData) => {
+                    if (groupFn || addSubRows) {
+                        const tableItem =
+                            rowData.props?.key && sorted.find((tableItem) => tableItem.key === rowData.props.key)
+                        if (tableItem) {
+                            action.click(tableItem.item)
+                        }
+                    } else {
+                        action.click(paged[rowId].item)
                     }
-                } else {
-                    rowAction.click(paged[rowId].item)
-                }
-            },
+                },
+            })
+        })
+        return actions
+    }
+
+    // Parse static actions
+    const actions = parseRowAction(rowActions)
+
+    // Wrap provided action resolver
+    let actionResolver: IActionsResolver | undefined
+    if (rowActionResolver) {
+        actionResolver = (rowData: IRowData) => {
+            let tableItem
+            if (groupFn || addSubRows) {
+                tableItem = rowData.props?.key && sorted.find((tableItem) => tableItem.key === rowData.props.key)
+            } else {
+                tableItem = paged[rowData.secretTableRowKeyId]
+            }
```

Comment:
```suggestion
            let tableItem
            if (groupFn || addSubRows) {
                tableItem = rowData.props?.key && paged.find((tableItem) => tableItem.key === rowData.props.key)
            } else if ((extraData?.rowIndex ?? -1) >= 0) {
                tableItem = paged[extraData?.rowIndex ?? 0]
            }
```

Same as for static actions, we can check just in the `paged` array. For the non-grouped/expandable case, I'm not sure if we should rely on `secretTableRowKeyId`. `IExtraData` has a `rowIndex` property which seems to work, but it is optional and I'm not really sure why.